docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst

```diff
@@ -0,0 +1,169 @@
+..
+   Copyright (c) 2023 OpenHW Group
+   Copyright (c) 2023 Thales
+
+   SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
+
+.. Level 1
+   =======
+
+   Level 2
+   -------
+
+   Level 3
+   ~~~~~~~
+
+   Level 4
+   ^^^^^^^
+
+.. _cva6_riscv_instructions_RVZcmp:
+
+*Applicability of this chapter to configurations:*
+
+.. csv-table::
+   :widths: auto
+   :align: left
+   :header: "Configuration", "Implementation"
+
+   "CV32A60AX", "Implemented extension"
+   "CV32A60X", "Implemented extension"
+
+**Note**: Zcmp is primarily targeted at embedded class CPUs due to implementation complexity. Additionally, it is not compatible with architecture class profiles.  
+
+RVZcmp Code Size Reduction Instructions
+---------------------------------------
+
+Zcmp belongs to group of extensions called RISC-V Code Size Reduction Extension (Zc*). Zc* has become the superset of Standard C extension adding more 16-bit instructions to the ISA.
+Zcmp includes 16-bit macro instructions, PUSH/POP and double move, which resue the encoding for c.fsdsp instruction.
+All the Zcmp instructions require at least standard C extension support as pre-requisite, along with Zca extension.
+
+- **CM.PUSH**: Compressed Push
+
+    **Format**: cm.push {reg_list}, -stack_adj
+
+    **Description**: This instruction pushes (stores) the registers in reg_list to the memory below the stack pointer, and then creates the stack frame by decrementing the stack pointer by stack_adj, including any additional stack space requested by the value of spimm.
+
+    **Pseudocode**: if (XLEN==32) bytes=4; else bytes=8;
+                    addr=sp-bytes;
+                    for(i in 27,26,25,24,23,22,21,20,19,18,9,8,1) {
+                        if (xreg_list[i]) {
+                            switch(bytes) {
+                                4: asm("sw x[i], 0(addr)");
+                                8: asm("sd x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    sp-=stack_adj;
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
```

Comment:
It is difficult to know what are the invalid values for `reg_list`. According to RISC-V spec, it is not only {ra, s0-s11} which is valid but also {ra}, {ra, s0}, {ra, s0-s1}, ..., {ra, s0-s9}.

Regarding stack_adj, the invalid values are depending on XLEN.